cns/service/main.go

```diff
@@ -1019,6 +1045,10 @@ func main() {
 		}(privateEndpoint, infravnet, nodeID)
 	}
 
+	if config.ChannelMode == cns.AzureHost {
+		httpRemoteRestService.StartNodeSubnet(rootCtx)
```

Comment:
why dont we call startnodesubnet after initializenodesubnet?

Reply:
We want to wait till the service http service is running, and we catch up on all async deletes.

Reply:
is async delete happen as part of http service start? i didnt understand clearly. can you add a comment why it should be started after http service start?

Reply:
Async delete is done by having the HTTP service running, and then submitting the pending delete requests to the service. Added a comment.

Reply:
couple of questions:
1. what happens if we start nodesubnet before async delete?
2. how do we make sure async delete is called before this call?

Reply:
1. I don't think there will be a major problem, I anticipate the pods pending deletion will stay in the in-memory state and then eventually deleted when async delete catche up
2. We do async delete a few lines before in this very function, before we call `StartNodeSubnet`

Reply:
@tamilmani1989 If you think I should start nodesubnet early, I could try that, but I suspect we might run into some other issues if the HTTP service isn't running yet. In my implementation so far I have always finished async delete before starting nodesubnet.

Reply:
@tamilmani1989 I tried starting nodesubnet early and it does work without errors. Then I recalled the other reason I did this - `StartNodeSubnet` will cause the conflist to be generated, which means CNS needs to be ready to handle IPAM queries. That doesn't happen until rest service is running. So conflist generation (and hence StartNodeSubnet) has to wait until then.